### ISSUE TITLE
Expanding vs sliding window splitter (window splitter with fixed start point)

### ISSUE THREAD

**Issue by juanitorduz** [alan-turing-institute/sktime#552]
**Is your feature request related to a problem? Please describe.**

As in real use cases we collect data to enlarge our training set, it would be useful to have an option on the `SlidingWindowSplitter` to have fixed start windows for the fold. For example for `window_length = 5`, `step_length = 1` and `fh = 3`
here is a representation of the folds:
```
|-----------------------|
| * * * * * x x x - - - |
| - * * * * * x x x - - |
| - - * * * * * x x x - |
| - - - * * * * * x x x |

* = training fold.
x = test fold.  
```

We could add the option to have the folds of the form:
```
|-----------------------|
| * * * * * x x x - - - |
| * * * * * * x x x - - |
| * * * * * * * x x x - |
| * * * * * * * * x x x |
```

**Describe the solution you'd like**
Just add an option on the `SlidingWindowSplitter` class to keep `start_fixed`.

**Describe alternatives you've considered**
When this parameter is set to `True` we compute the folds as 
```python
end = self._get_end(y)
start = self._get_start()

for split_point in range(start, end, step_length):
            training_window = np.arange(start, split_point)
            test_window = split_point + fh - 1
            yield training_window, test_window
```

**Additional context**
 See `fixedWndow` parameter from [Caret](https://topepo.github.io/caret/data-splitting.html#data-splitting-for-time-series)

![image](https://user-images.githubusercontent.com/22996444/102721496-65f96600-42fb-11eb-9f67-fb4bc446592b.png)

**Comment by aiwalter** [comment]
great idea @juanitorduz. I think that is what is usually called "expanding window" ? Mixing this up with the `SlidingWindowSplitter` class might be a bit confusing in naming? I would propose to either rename this class to just `WindowSplitter` then with an argument `mode: ["sliding", "expanding"]` or a new class `ExpandingWindowSplitter`?

**Comment by juanitorduz** [comment]
> great idea @juanitorduz. I think that is what is usually called "expanding window" ? Mixing this up with the `SlidingWindowSplitter` class might be a bit confusing in naming? I would propose to either rename this class to just `WindowSplitter` then with an argument `mode: ["sliding", "expanding"]` or a new class `ExpandingWindowSplitter`?

Yes, this naming makes sense 👍

**Comment by mloning** [comment]
Thanks for the suggestion @juanitorduz! This is related to #520. 

This may be obvious, but I'm always a little confused by the window splitter. Since our forecasters remember the training data they see, all of them are technically working with an expanding window, no? When would you use an `ExpandingWindowSplitter`?

**Comment by juanitorduz** [comment]
Ups! Sorry for the duplicated issue, i was filtering by the keyword `SlidingWindowSplitter`. 

In addition to @aiwalter 's input in https://github.com/alan-turing-institute/sktime/issues/520 (thanks for that!) I have some additional comment: In many practical situations I need the expanded-folds themselves. For example, when doing back testing: Suppose I need to run a forecast model on a daily basis (we train and predict on a daily basis) and I want to study the performance of an already selected model (let us say ARIMA(p=1, i=0, q=1), i.e. I do not need to select any parameters). Then I would like to train and predict on the expanded-folds to do an error analysis and understand what to expect of the model when deployed to production. 
Does it make sense? 😬 Is there a way of doing this with sktimie 0.5.0?

Scikit-Learn has already somethinig similar: [TimeSeriesSplit](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html#sklearn.model_selection.TimeSeriesSplit), but id does not offer as many options as `SlidingWindowSplitter`.

**Comment by fkiraly** [comment]
@mloning, I agree with @juanitorduz - sliding and expanding window training/forecasting are in-principle different. The sliding window splitter predicts only from some recent part of the available training data - this can make sense if there is non-stationarity or concept drift and you do not want to "contaminate" the forecaster with older data.
@mloning, the additional point you raise is "what the method does" vs "what the method has access to, given its template". I think the template is designed to have access to all past data, but we can of course make specific methods which don't use it, e.g., due to the non-stationarity issue above.

Regarding @aiwalter's suggestion, i.e., whether the expanding window should be a modification of `SlidingWindowSplitter`, I'm not sure.
For one, it makes sense, since you could simply re-train and forget the oldest data, or not, and that can be part of the same reduction strategy.

However, there is a more general wrapper strategy, which forces the wrapped forecaster to forget the oldest data. The wrapped forecaster would not have to be a strategy that reduces to supervised prediction, but could be any forecaster. Not sure whether it is useful or too general though?

**Comment by fkiraly** [comment]
@juanitorduz, one additional point perhaps on your comment: we should be careful and not mix evaluation and tuning. Tuning are wrappers/compositors like `GridSearchCV` in `sklearn`; evaluation should be external to any estimator. We are still working on the evaluation/monitoring module - let us know if you are interested to contribute.

**Comment by juanitorduz** [comment]
You are right @fkiraly! We should indeed be careful in the distinction between tuning and evaluation! Having an evaluation/monitoring module would be really helpful! For example, I find  [Prophet's diagnostics](https://facebook.github.io/prophet/docs/diagnostics.html) quite handy. 

I would be interested in contributing, although my capacity is limited at the moment (baby at home 👶 ). I could support with small bugs/features and also testing module designs 🤓 .

**Comment by koralturkk** [comment]
@fkiraly I would like to contribute to implementing this feature. I would like to ask for your opinion on how this feature should be implemented by an ExpandingWindowSplitter additionally or by creating just a WindowSplitter  that has an arg e.g. splitter_type: ["sliding", "expanding"] as suggested before?

**Comment by juanitorduz** [comment]
Thank you @koralturkk for https://github.com/alan-turing-institute/sktime/pull/627 :)

**Comment by mloning** [comment]
Closed by #627 - thanks @koralturkk - much appreciated! 🎉
